xonsh/xoreutils/uptime.py

```diff
@@ -86,10 +86,16 @@ def _uptime_amiga():
 
 def _uptime_beos():
     """Returns uptime in seconds on None, on BeOS/Haiku."""
-    if not hasattr(xp.LIBC, "system_time"):
+    try:
+        libroot = ctypes.CDLL("libroot.so")
```

Comment:
It would be good to move this to `xonsh.platform`, so other tools can use the same object.